src/event.rs

```diff
@@ -53,6 +54,37 @@ impl Event {
         unsafe { ffi::gdk_event_put(self.to_glib_none().0) }
     }
 
+    /// Set the event handler.
+    /// 
+    /// The callback `func` is called for each event. If `None`, event handling
+    /// is disabled.
+    /// 
+    /// The `data` pointer is passed to `func` and `notify` when called.
+    /// 
+    /// Should another event handler be set in the future, `notify` will be
+    /// called before doing so, if not `None`.
+    pub fn set_handler<T>(
+        func: Option<&'static Fn(Event, Option<*mut T>)>,
```

Comment:
Second parameter in closure IMHO wrong, and don't need in rust closures at all, better always pass `data=null_ptr`